apps/emqx_conf/src/emqx_conf_schema.erl

```diff
@@ -1105,12 +1105,7 @@ translation("gen_rpc") ->
     [{"default_client_driver", fun tr_default_config_driver/1}];
 translation("prometheus") ->
     [
-        {"vm_dist_collector_metrics", fun tr_vm_dist_collector/1},
-        {"mnesia_collector_metrics", fun tr_mnesia_collector/1},
-        {"vm_statistics_collector_metrics", fun tr_vm_statistics_collector/1},
-        {"vm_system_info_collector_metrics", fun tr_vm_system_info_collector/1},
-        {"vm_memory_collector_metrics", fun tr_vm_memory_collector/1},
-        {"vm_msacc_collector_metrics", fun tr_vm_msacc_collector/1}
```

Comment:
Previous:
we set collector's metrics to empty `[]` or `all` to disabled/enabled collector.
But it's will run collector even metrics is [].
So we now remove the collector when it's disabled. It's better than previous.